### DISCUSSION TITLE
Oauth?

### DISCUSSION THREAD

**Discussion by stevenbaert** [ItzCrazyKns/Vane#356] [Ideas]
Hi,

Fantastic project!

The only thing stopping me from fully adopting it is the security aspect. Perplexity supports user settings and OAuth. It would be great if Perplexica offered similar features! This way, I could place it behind my reverse proxy and access it securely from anywhere only offering oauth (avoiding brute force password attempts) 😁

Looking forward to your thoughts.

Kind regards,
Steven

**Comment by papiche** [comment]
Facing the same problem,

Perplexica runs in my LAN, but "no access control" is made, so I cannot open this service to the Web.
I solved it with a trick : allowing ssh reverse tunnel from my users (which public key is placed in  ```~/.ssh/authorized_keys```)


Then this script is run by remote users to get access to Perplexica

```
#!/bin/bash

echo "Looking for already open tunnel"
tunnelPID=$(lsof -i :5000 | grep ssh | xargs | cut -d ' ' -f 2)
[[ ! -z $tunnelPID ]] && kill -9 $tunnelPID && echo "5000 port found"
tunnelPID=$(lsof -i :3001 | grep ssh | xargs | cut -d ' ' -f 2)
[[ ! -z $tunnelPID ]] && kill -9 $tunnelPID && echo "3001 port found"

echo "Reverse Tunnels"
echo "Perplexica"
ssh -fN -L 5000:127.0.0.1:3000 me@myIP -p 22222
echo "Perplexica API"
ssh -fN -L 3001:127.0.0.1:3001 me@myIP -p 22222

xdg-open http://127.0.0.1:5000
```
NB: activate fail2ban & forward port 22222 to 22 on LAN router

**Comment by bpawnzZ** [comment]
Setup cloudflare "ZeroTrust/SafeAccess" for access control..  You can setup Oauth through github or google.  i been using this to secure my services for probably 2 years now. as long as you are logged into said account its invisible essentially.  It reallly is the perfect solution for what you are asking for

![0dea2](https://github.com/user-attachments/assets/350b214c-3d4f-4084-b61e-49e92b284a59)
directly behind that login is Perplexica.  This form of Oauth can be used to protect any service you like however.  Its a good robust solution to use for alof of services.  Not just Perplexica.. its worth knowing

**Reply by stevenbaert** [reply]
Awesome! Can you clarify what subscription is needed and if you know a setup guide that would be handy. Thanks

**Reply by bpawnzZ** [reply]
No subscription.. Its 100% free. Free plan.  I'll try to put together a guide of some sort later this afternoon.  I got to make my employer money for the next few hrs. Lol

**Reply by stevenbaert** [reply]
Thanks! Up and running now, though the page does not load fully. Could you share any specific settings for Cloudflare and Perplexica?

**Reply by bpawnzZ** [reply]
Your done with the Oath part.  as long as you get the ZeroTrust login and you can successfully login and it redirects your done with cloudflare.  However you will need to rebuild project setup correctly with your reverse proxy.  I posted about a month ago how to setup with Traefik and responded to someone yesterday about how to setup with NGINX.  Let me know if you cant figure it out.  

[Traefik Link](https://github.com/ItzCrazyKns/Perplexica/discussions/457)

[Nginx Link](https://github.com/ItzCrazyKns/Perplexica/discussions/477)

**Reply by stevenbaert** [reply]
Don't get that. Via the nginx internal proxy address Perplexica works. 
But then you have no oauth of course. That's why I was looking for Cloudflare.

Why would you use proxy if you have Cloudflare? My Perplexica page does load only not fully (hangs since it cannot reach something).

**Reply by bpawnzZ** [reply]
Try setting up the settings for perplexica when you first have the page load.  input base_url mdoel etc.  save and reload.

**Reply by stevenbaert** [reply]
Thanks but again no clue what you mean.

**Reply by fakebizprez** [reply]
@stevenbaert what he means is you don't need a reverse proxy if you're using Cloudflare because you can just use a Cloudflared Tunnel.
